server/src/main/java/io/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -653,27 +654,70 @@ private void stopBeingLeader()
     if (config.isConvertSegments()) {
       helpers.add(new DruidCoordinatorVersionConverter(indexingServiceClient, whitelistRef));
     }
-    if (config.isMergeSegments()) {
-      helpers.add(new DruidCoordinatorSegmentMerger(indexingServiceClient, whitelistRef));
-      helpers.add(
-          new DruidCoordinatorHelper()
-          {
-            @Override
-            public DruidCoordinatorRuntimeParams run(DruidCoordinatorRuntimeParams params)
-            {
-              CoordinatorStats stats = params.getCoordinatorStats();
-              log.info("Issued merge requests for %s segments", stats.getGlobalStats().get("mergedCount").get());
 
-              params.getEmitter().emit(
-                  new ServiceMetricEvent.Builder().build(
-                      "coordinator/merge/count", stats.getGlobalStats().get("mergedCount")
-                  )
-              );
+    final String mergeStrategy = config.getMergeStrategy();
 
-              return params;
-            }
-          }
-      );
+    if (mergeStrategy != null) {
+
+      switch (mergeStrategy) {
+        case "append": {
```

Comment:
Most other places have this kind of logic handled through jackson type bindings. Can that approach be done here?